Tests/SourceKitLSPTests/TypeHierarchyTests.swift

```diff
@@ -0,0 +1,167 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import ISDBTestSupport
+import LanguageServerProtocol
+import XCTest
+
+final class TypeHierarchyTests: XCTestCase {
+  func testTypeHierarchy() throws {
+    let ws = try staticSourceKitTibsWorkspace(name: "TypeHierarchy")!
+    try ws.buildAndIndex()
+
+    try ws.openDocument(ws.testLoc("a.swift").url, language: .swift)
+
+    // Requests
+
+    func typeHierarchy(at testLoc: TestLocation) throws -> [TypeHierarchyItem] {
+      let textDocument = testLoc.docIdentifier
+      let request = TypeHierarchyPrepareRequest(textDocument: textDocument, position: Position(testLoc))
+      let items = try ws.sk.sendSync(request)
+      return items ?? []
+    }
+
+    func supertypes(at testLoc: TestLocation) throws -> [TypeHierarchyItem] {
+      guard let item = try typeHierarchy(at: testLoc).first else {
+        fatalError("Type hierarchy at \(testLoc) was empty")
+      }
+      let request = TypeHierarchySupertypesRequest(item: item)
+      let types = try ws.sk.sendSync(request)
+      return types ?? []
+    }
+
+    func subtypes(at testLoc: TestLocation) throws -> [TypeHierarchyItem] {
+      guard let item = try typeHierarchy(at: testLoc).first else {
+        fatalError("Type hierarchy at \(testLoc) was empty")
+      }
+      let request = TypeHierarchySubtypesRequest(item: item)
+      let types = try ws.sk.sendSync(request)
+      return types ?? []
+    }
+
+    // Convenience functions
+
+    func testLoc(_ name: String) -> TestLocation {
+      ws.testLoc(name)
+    }  
+
+    func loc(_ name: String) -> Location {
+      Location(badUTF16: ws.testLoc(name))
+    }
+
+    func withoutData(_ item: TypeHierarchyItem) -> TypeHierarchyItem {
+      var item = item
+      item.data = nil
+      return item
+    }
+
+    /// Compares the given type hierarchies ignoring the implementation-specific
+    /// data field (which includes e.g. USRs that are difficult to test, especially
+    /// in the presence of extensions, and are not user-visible anyway).
+    func assertEqualIgnoringData(
+      _ actual: [TypeHierarchyItem],
+      _ expected: [TypeHierarchyItem],
+      file: StaticString = #file,
+      line: UInt = #line
+    ) {
+      XCTAssertEqual(
+        actual.map(withoutData),
+        expected.map(withoutData),
+        file: file,
+        line: line
+      )
```

Comment:
We ignore the LSP-specific `data` field in comparisons since its USRs are hard to test in the presence of extensions (which have individual USRs) and are not user-facing anyway. Thus we consider the `data` field as an implementation detail here that the test treats as a black box.